storm-core/src/jvm/org/apache/storm/daemon/Task.java

```diff
@@ -179,23 +180,22 @@ public BuiltinMetrics getBuiltInMetrics() {
     private TopologyContext mkTopologyContext(StormTopology topology) throws IOException {
         Map conf = (Map) workerData.get("conf");
         return new TopologyContext(
-            topology,
-            (Map) workerData.get("storm-conf"),
-            (Map<Integer, String>) workerData.get("task->component"),
-            (Map<String, List<Integer>>) workerData.get("component->sorted-tasks"),
-            (Map<String, Map<String, Fields>>) workerData.get("component->stream->fields"),
-            (String) workerData.get("storm-id"),
-            ConfigUtils.supervisorStormResourcesPath(ConfigUtils.supervisorStormDistRoot(conf, (String) workerData.get("storm-id"))),
-            ConfigUtils.workerPidsRoot(conf, (String) workerData.get("worker-id")),
-            taskId,
-            (Integer) workerData.get("port"),
-            (List<Integer>) workerData.get("task-ids"),
-            (Map<String, Object>) workerData.get("default-shared-resources"),
-            (Map<String, Object>) workerData.get("user-shared-resources"),
-            (Map<String, Object>) executorData.get("shared-executor-data"),
-            (Map<Integer, Map<Integer, Map<String, IMetric>>>) executorData.get("interval->task->metric-registry"),
-            (clojure.lang.Atom) executorData.get("open-or-prepare-was-called?")
-        );
+                topology,
+                (Map) workerData.get("storm-conf"),
```

Comment:
can we move these strings into final String at the top level. I prefer if we add a file ExecutorDataConstants and all these there.